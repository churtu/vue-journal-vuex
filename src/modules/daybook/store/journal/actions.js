import journalAPI from '@/api/journalAPI'


export const loadEntries = async ( { commit }) => {
    const { data } = await journalAPI.get('/entries.json')
    if(!data){
        commit('setEntries', [])
        return
    }
    const entries = []
     
    for( let id of Object.keys( data )){
        entries.push({
            id,
            ...data[id]
        })
    }
    commit('setEntries', entries)
}

export const updateEntry = async ( { commit }, entry) =>{
    const { picture, date, text, id } = entry 
    const dataToSave = {date, text, picture}
    await journalAPI.put(`/entries/${id}.json`, dataToSave)
    dataToSave.id = id
    commit('updateEntry', { ...dataToSave })
    
}

export const createEntry = async ( { commit }, entry ) => {
    const { text, date, picture } = entry
    const dataToSave = {date, text, picture}
    const { data } = await journalAPI.post('/entries.json', dataToSave)
    commit('addEntry', {text, date, id: data.name})
    return data.name // id en firebase
}

export const deleteEntry = async ( { commit }, id ) => {
    await journalAPI.delete(`/entries/${id}.json`)
    commit('deleteEntry', id)
}