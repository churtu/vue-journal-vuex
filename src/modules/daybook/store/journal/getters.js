
// export const myGetter = ( /*state*/ ) => {

// }

export const getEntiesByTerm = ( /*state*/ ) => {

}

export const getEntryById = ( /*state*/ ) => {

}