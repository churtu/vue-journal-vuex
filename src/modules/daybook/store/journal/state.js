

export default () => ({
    isLoading:true,
    entries: [
        {
            id: new Date().getTime(),
            date: new Date().toDateString(),
            text: 'Voluptate consectetur minim quis esse duis laborum dolore Lorem in cupidatat aliquip consequat nisi amet. Adipisicing nisi consequat consequat ad amet commodo. Eiusmod id laboris consequat aliqua veniam consectetur tempor. Officia aliquip veniam pariatur nostrud aliquip labore ad Lorem sint ea officia. Sint velit quis tempor excepteur occaecat est pariatur.',
            picture: null
        },
        {
            id: new Date().getTime()+1000,
            date: new Date().toDateString(),
            text: 'Voluptate consectetur minim quis esse duis laborum dolore Lorem in cupidatat aliquip consequat nisi amet. Adipisicing nisi consequat consequat ad amet commodo. Eiusmod id laboris consequat aliqua veniam consectetur tempor. Officia aliquip veniam pariatur nostrud aliquip labore ad Lorem sint ea officia. Sint velit quis tempor excepteur occaecat est pariatur.',
            picture: null
        },
        {
            id: new Date().getTime()+2000,
            date: new Date().toDateString(),
            text: 'Voluptate consectetur minim quis esse duis laborum dolore Lorem in cupidatat aliquip consequat nisi amet. Adipisicing nisi consequat consequat ad amet commodo. Eiusmod id laboris consequat aliqua veniam consectetur tempor. Officia aliquip veniam pariatur nostrud aliquip labore ad Lorem sint ea officia. Sint velit quis tempor excepteur occaecat est pariatur.',
            picture: null
        },
    ]
})