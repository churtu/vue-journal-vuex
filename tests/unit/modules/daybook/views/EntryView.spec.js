import { shallowMount } from '@vue/test-utils'
import { createStore } from 'vuex'
import journal from '@/modules/daybook/store/journal'
import { journalState } from '../../../mock-data/test-journal-state'
import EntryView from '@/modules/daybook/views/EntryView'
import Swal from 'sweetalert2'

const createVuexStore = ( initialState ) => createStore({
    modules: {
        journal: {
            ...journal,
            state: {...initialState}
        }
    }    
})

jest.mock('sweetalert2', () => ({
    fire: jest.fn(),
    showLoading: jest.fn(),
    close: jest.fn()
}))
describe('Pruebas en el EntryView', () => {
    const routerMock = {
        push: jest.fn()
    }
    const store = createVuexStore( journalState )
    store.dispatch = jest.fn() // para escuchar cuando se llama una action desde un componente
    let wrapper
    
    beforeEach(() => {
        jest.clearAllMocks()
        wrapper = shallowMount( EntryView, {
            props: {
                id: journalState.entries[0].id
            },
            global:{
                mocks: {
                    $router: routerMock
                },
                plugins: [store]
            }
        })
    })
    test('Debe sacar al usuario por que el id no existe', () => { 
        wrapper = shallowMount( EntryView, {
            global:{
                mocks: {
                    $router: routerMock
                },
                plugins: [store]
            },
            props: {
                id: 'No existe'
            }
        })

        expect( routerMock.push ).toHaveBeenCalledWith({name:'no-entry'})
     })

     test('Debe de mostrar la entrada correctamente', () => {
        expect(wrapper.html()).toMatchSnapshot()
        expect(routerMock.push).not.toHaveBeenCalled()
     })

     test('Debe de borrar la entrada y salida', async() =>{
         Swal.fire.mockReturnValueOnce(Promise.resolve({ isConfirmed: true}))
         await wrapper.find('.btn-danger').trigger('click')
         expect(Swal.fire).toHaveBeenCalledWith({
            title: 'Est√° seguro?',
            text: 'Una vez borrado, no se puede recuperar',
            showDenyButton: true,
            confirmButtonText: 'Se, estoy seguro'
          })
          expect(routerMock.push).toHaveBeenCalledWith({name: 'no-entry'})
          expect(store.dispatch).toHaveBeenCalledWith('journal/deleteEntry', '-MuvDbLfNDBGis9af7rj')
     })
})