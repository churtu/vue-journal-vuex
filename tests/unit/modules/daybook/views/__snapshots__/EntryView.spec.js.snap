// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el EntryView Debe de mostrar la entrada correctamente 1`] = `
<!-- template no cambia el html, sirve para agrupar condicionales -->
<div class="entry-title d-flex justify-content-between p-2">
  <div><span class="text-success fs-3 fw-bold">2</span><span class="mx-1 fs-3">Febrero</span><span class="mx-2 fs-4 fw-light">2022, Miércoles</span></div>
  <div><input type="file" style="display: none;"><button class="btn btn-danger mx-2"> Borrar <i class="fa fa-trash-alt"></i></button><button class="btn btn-primary"> Subir foto <i class="fa fa-upload"></i></button></div>
</div>
<hr>
<div class="d-flex flex-column px-3 h-75"><textarea placeholder="¿Que sucedio hoy?"></textarea></div>
<!-- <img
    src="https://cdn0.matrimonios.cl/vendor/4636/3_2/960/jpg/ideal-para-tu-evento.jpeg"
    alt="image"
    class="img-thumbnail"
  /> -->
<!--v-if-->
<!--v-if-->
<fab-stub icon="fa-save"></fab-stub>
`;
